### PR TITLE
bump documenter version, use doc pkg cache

### PR DESCRIPTION
**Pull request by exaexa**

### CODE REVIEW

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -16,6 +16,7 @@ jobs:
     steps:
       - uses: actions/checkout@v3
```
Comment (giordano) [inline .github/workflows/docs.yml:17]:
```suggestion
      - uses: actions/checkout@v4
```
Related, it'd be good to add dependatbot to automate this: https://github.com/JuliaPhysics/Measurements.jl/blob/21cf93c7dc25b7a1b14d5b877bc8f4884599ff5e/.github/dependabot.yml

**Review by giordano — APPROVED** [review]
Looks good to me if it works :+1:

File: .github/workflows/docs.yml (hunk 1 above)
Comment (exaexa) [reply .github/workflows/docs.yml:17]: hm true. I actually thought I'd make a common repo of all CI scriptage and update from there, dependabit might work too. Will have a look asap, thanks.

Comment (exaexa) [reply .github/workflows/docs.yml]: (applied manually)

**Pull request merged**